### PR TITLE
chore: extract control locator

### PR DESCRIPTION
**Pull request by tsaleksandrova**

### CODE REVIEW

File: src/locators/defaultLocators.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  // http://angular.github.io/protractor/#/api?view=ProtractorBy.prototype.addLocator
+
+  by.addLocator('jq', function(query,opt_parentElement) {
+    return $(opt_parentElement ? opt_parentElement + ' ' + query : query).toArray();
+  });
+
+  var byControlLocator = new ByControlLocator(clientsidescripts, this.logger);
+  byControlLocator.register(by, 'control');
+
+  /* example of how to register a new locator that:
```
Comment (maximnaidenov) [inline src/locators/defaultLocators.js]: I  think it is better idea to have the new locator as a module that will use our byControl locator module.

File: src/locators/defaultLocators.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Register jasmine locator
+ * @param {By} by - jasmine By object on which to add the new locator
+ */
+DefaultLocators.prototype.register = function(by) {
+  this.logger.debug('Registering default locators');
+  // http://angular.github.io/protractor/#/api?view=ProtractorBy.prototype.addLocator
+
+  by.addLocator('jq', function(query,opt_parentElement) {
```
Comment (maximnaidenov) [inline src/locators/defaultLocators.js]: maybe it is time to externalize this also in a separate module.

File: src/locators/defaultLocators.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+var clientsidescripts = require('../scripts/clientsidescripts');
```
Comment (maximnaidenov) [inline src/locators/defaultLocators.js]: This module tends to grow too much, lets remove it and add the default locators in the profile.conf.js, similar to the reporters.

File: src/locators/byControlLocator.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    }.bind(this));
+};
+
+/**
+ * get a string representation of the locator for logging purposes
+ * @returns {string}
+ */
+_ControlLocator.prototype.toString = function toString() {
+  var sMatchers = JSON.stringify(this.matchers);
+  return 'by.control(' + sMatchers + ')';
```
Comment (maximnaidenov) [inline src/locators/byControlLocator.js]: should use the specified name from the constructor

File: src/locators/byControlLocator.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+var _ = require('lodash');
+
+/**
+ * Control locator public class
+ * @param {object} clientsidescripts object containing defnition of functions to be executed in the browser
+ * Expected functions are findByControl and getLatestLog.
+ * @param {object} logger optional logger object with methods: debug, info, error, trace. if not defined, no logs will be produced
+ * @public
+ */
+function ControlLocator(clientsidescripts, logger) {
```
Comment (maximnaidenov) [inline src/locators/byControlLocator.js]: you can get the clientside scripts as module, no need to pass a ref here

File: src/locators/customLocator.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    // mOPA5Matchers = {controlType: "sap.m.Button"}
+    var mOPA5Matchers = {};
+    if (mMatchers.elementProperties && mMatchers.elementProperties.metadata) {
+      mOPA5Matchers.controlType = mMatchers.elementProperties.metadata;
+    }
+    var byControlLocator = new ControlLocator(this.logger);
+    return byControlLocator.apply(mOPA5Matchers);
+  };
+};
+
```
Comment (maximnaidenov) [inline src/locators/customLocator.js]: what about the .toString with the correct name ?

File: src/locators/customLocator.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
+var ControlLocator = require('./controlLocator');
```
Comment (maximnaidenov) [inline src/locators/customLocator.js]: At the end, this file should go into a separate PR that Georgios will use as a sample. Or maybe move as commented section in the byControl ?

File: src/locators/jqueryLocator.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+ * Register jasmine locator
+ * @param {By} by - jasmine By object on which to add the new locator
+ */
+JQueryLocator.prototype.register = function (by) {
+  this.logger.debug('Registering jquery locator');
+  // http://angular.github.io/protractor/#/api?view=ProtractorBy.prototype.addLocator
+
+  by.addLocator('jq', function (query, opt_parentElement) {
```
Comment (maximnaidenov) [inline src/locators/jqueryLocator.js]: what about reusing the pattern with registering the controlLocator and so supporting the .toString() ?

File: src/locators/customLocator.js (hunk 6 above)
Comment (tsaleksandrova) [reply src/locators/customLocator.js]: the name isn't configurable

Comment (maximnaidenov) [reply src/locators/customLocator.js]: but if the custom locator overides the .toString() it will be used in error messages ?

File: src/locators/jqueryLocator.js (hunk 8 above)
Comment (tsaleksandrova) [reply src/locators/jqueryLocator.js]: there is a default toString impl

File: src/locators/customLocator.js (hunk 7 above)
Comment (tsaleksandrova) [reply src/locators/customLocator.js]: moving it to a new pr

File: src/locators/byControlLocator.js (hunk 4 above)
Comment (tsaleksandrova) [reply src/locators/byControlLocator.js]: the name isn't configurable anymove
